mlir/tools/mlir-miopen-driver/mlir-miopen-lib.cpp

```diff
@@ -133,4 +149,73 @@ extern "C" const char *MlirGenIgemmCflags(MlirHandle mlirHandle) {
   translateModuleToMIOpenCFlags(handle->getModule(), handle->genTxt);
   return (handle->genTxt).c_str();
 }
+
+extern "C" void MlirLowerBin(MlirHandle mlirHandle) {
+  MlirHandle_s *handle = static_cast<MlirHandle_s *>(mlirHandle);
+  ModuleOp module = handle->getModule();
+
+  llvm::InitializeAllTargetInfos();
+  llvm::InitializeAllTargetMCs();
+  llvm::InitializeAllAsmParsers();
+
+  // Initialize LLVM AMDGPU backend.
+  LLVMInitializeAMDGPUTarget();
+  LLVMInitializeAMDGPUTargetInfo();
+  LLVMInitializeAMDGPUTargetMC();
+  LLVMInitializeAMDGPUAsmPrinter();
+  mlir::initializeLLVMPasses();
+
+  PassManager pm(module.getContext());
+
+  BackendUtils utils;
+
+  // Passes for lowering MIOpen dialect.
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep1Pass());
+  pm.addPass(mlir::miopen::createAffineTransformPass());
+  pm.addPass(mlir::miopen::createAffixTuningParametersPass(0, 0));
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep2Pass());
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep3Pass());
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep4Pass());
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep5Pass());
+  pm.addPass(
+      mlir::createLowerMIOpenOpsToGPUPass("miopen_conv2d_kcyx_nchw_nkhw"));
```

Comment:
From MIOpen's perspective, they just need a hard-coded kernel name. However, from API perspective, we might do one of the following:
- Let the user decide the kernel name in creation of the handle
- Use the same convention as `mlir-miopen-driver` to give it a hard-coded name

Reply:
since MIOpen may generate kernels in different layouts very soon, probably better to follow the same convention with `mlir-miopen-driver.`

Reply:
Done in bc7d022:
- When user pass in a non-empty kernel name, it uses that name 
- When kernel name is passed in as an empty string, it falls back to existing naming convention